guide_drafts/model-derives.md

```diff
@@ -0,0 +1,935 @@
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `diesel_demo::models::Post : diesel::associations::HasTable`
+           `diesel_demo::models::Post : diesel::associations::HasTable`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&diesel_demo::models::Post : diesel::associations::HasTable`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<_>`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `diesel_demo::models::Post : diesel::associations::HasTable`
+           `diesel_demo::models::Post : diesel::associations::HasTable`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&mut diesel_demo::models::Post : diesel::BelongingToDsl<&[_]>`
+           `&mut diesel_demo::models::Post : diesel::associations::HasTable`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<_>`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&mut diesel_demo::models::Post : diesel::associations::HasTable`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&mut diesel_demo::models::Post : diesel::associations::HasTable`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<_>`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&mut diesel_demo::models::Post : diesel::associations::HasTable`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+           `&mut diesel_demo::models::Post : diesel::associations::BelongsTo<diesel_demo::models::User>`
+```
+
+In this particular case,
+we are using the `belongs_to()` method, which requires us to implement the `BelongsTo` trait.
+Below is what `cargo expand` generates for `#[derive(Associations)]`.
+
+```rust
+#[derive(Identifiable, Queryable)]
+pub struct User {
+    pub id: i32,
+    pub first_name: String,
+    pub last_name: String,
+    pub email: Option<String>,
+}
+
+#[derive(Insertable)]
+#[table_name="users"]
+pub struct NewUser<'a> {
+    pub first_name: &'a str,
+    pub last_name: &'a str,
+    pub email: Option<&'a str>,
+ }
+
+#[derive(AsChangeset)]
+#[table_name="users"]
+pub struct UpdateUser<'a> {
+    pub first_name: &'a str,
+    pub last_name: &'a str,
+    pub email: Option<Option<&'a str>>,
+ }
```

Comment:
I you use the "let's just update the name because email requires double opt-in" use case it makes more sense because then the struct is _not_ the same as `NewUser` :)

Reply:
Ah, forget it, you can't show the `Option<Option<_>>` then. Sorry, commented too fast.